lesson_13/adrian_filimon/lesson_13.py

```diff
@@ -0,0 +1,64 @@
+from math import sqrt
+
+class PerimeterMixin:
+    def perimeter(self):
+
+        class_name = type(self).__name__
+
+        return f"This is the perimeter for {class_name}: {sum(self.sides)}"
+
+
+class Polygon(PerimeterMixin):
+    def __init__(self, *args):
+        self.sides = args
+
+    def __str__(self):
+        no_of_sides = len(self.sides)
+        return f"{no_of_sides} is the number of sides."
+
+    def display(self):
+        for side_index, length in enumerate(self.sides, start=1):
+            print(f"Side {side_index} with length: {length}")
+
+
+class Triangle(Polygon):
+    # `s(s-a)(s-b)(s-c) ** 0.5` where s = `(a+b+c) / 2`
+    def __init__(self, *args):
+        super().__init__(*args)
+
+    def area(self):
+        a, b, c = self.sides
+        s = sum(self.sides) / 2
+        return s * (s - a) * (s - b) * (s - c) ** 0.5
+
+
+class Square(Polygon):
+    def __init__(self, *args):
+        super().__init__(*args)
+
+    def area(self):
+        side, *_ = self.sides
+        return side ** 2
+
+    def from_area(area):
```

Comment:
This method is supposed to return a new Square object. The logic of getting the sides is correct, but with the sides, we need to create a new object. In order to create a new object we need the class object (cls).

What we can use from python to achieve this?